device/cuda/src/seeding/spacepoint_binning.cu

```diff
@@ -0,0 +1,92 @@
+
+// Project include(s).
+#include "traccc/seeding/device/count_grid_capacities.hpp"
+#include "traccc/seeding/device/get_spacepoint_prefix_sum.hpp"
+#include "traccc/seeding/device/populate_grid.hpp"
+
+namespace traccc::cuda {
+namespace kernels {
+
+/// CUDA kernel for running @c traccc::device::count_grid_capacities
+__global__ void count_grid_capacities(
+    seedfinder_config config, sp_grid::axis_p0_type phi_axis,
+    sp_grid::axis_p1_type z_axis, spacepoint_container_const_view spacepoints,
+    vecmem::data::vector_view<const device::get_spacepoint_prefix_sum_element_t>
+        sp_prefix_sum,
+    vecmem::data::vector_view<unsigned int> grid_capacities) {
+
+    device::count_grid_capacities(threadIdx.x + blockIdx.x * blockDim.x, config,
+                                  phi_axis, z_axis, spacepoints, sp_prefix_sum,
+                                  grid_capacities);
+}
+
+/// CUDA kernel for running @c traccc::device::populate_grid
+__global__ void populate_grid(
+    seedfinder_config config, spacepoint_container_const_view spacepoints,
+    vecmem::data::vector_view<const device::get_spacepoint_prefix_sum_element_t>
+        sp_prefix_sum,
+    sp_grid_view grid) {
+
+    device::populate_grid(threadIdx.x + blockIdx.x * blockDim.x, config,
+                          spacepoints, sp_prefix_sum, grid);
+}
+
+}  // namespace kernels
+
+spacepoint_binning::spacepoint_binning(
+    const seedfinder_config& config, const spacepoint_grid_config& grid_config,
+    vecmem::memory_resource& mr)
+    : m_config(config), m_axes(get_axes(grid_config, mr)), m_mr(mr) {}
+
+sp_grid_buffer spacepoint_binning::operator()(
+    const host_spacepoint_container& spacepoints) const {
+
+    // Get the prefix sum for the spacepoints.
+    device::get_spacepoint_prefix_sum_result_t sp_prefix_sum =
+        device::get_spacepoint_prefix_sum(spacepoints, m_mr);
+
+    // Set up the container that will be filled with the required capacities for
+    // the spacepoint grid.
+    const std::size_t grid_bins = m_axes.first.n_bins * m_axes.second.n_bins;
+    vecmem::vector<unsigned int> grid_capacities(grid_bins, 0, &m_mr.get());
+
+    // Calculate the number of threads and thread blocks to run the kernels for.
+    const unsigned int num_threads = WARP_SIZE * 8;
+    const unsigned int num_blocks = sp_prefix_sum.size() / num_threads + 1;
+
+    // Fill the grid capacity container.
+    kernels::count_grid_capacities<<<num_blocks, num_threads>>>(
+        m_config, m_axes.first, m_axes.second, traccc::get_data(spacepoints),
+        vecmem::get_data(sp_prefix_sum), vecmem::get_data(grid_capacities));
+    CUDA_ERROR_CHECK(cudaGetLastError());
+    CUDA_ERROR_CHECK(cudaDeviceSynchronize());
+
+    // Create the grid buffer.
+    sp_grid_buffer grid_buffer(m_axes.first, m_axes.second,
+                               std::vector<std::size_t>(grid_bins, 0),
+                               std::vector<std::size_t>(grid_capacities.begin(),
```

Comment:
Hmm you removed the minimum capacity of 60 that I added in #160. 
I don't know if it's a bug in vecmem, the small pileup ttbar sample (e.g. ttbar<40>) made run time error.

Reply:
Kudos for noticing!

Yes, I removed that hack, hoping to bump into the issue that made you add it. I think you're right, there may very well be an issue with the VecMem code. @konradkusiak97 saw something very similar in his code, where he ended up adding +1 to every size to avoid a crash.

Thanks for the hint about the mu40 samples. I'll play around with those a bit.